client/post-editor/media-modal/test/index.jsx

```diff
@@ -85,194 +76,229 @@ describe( 'EditorMediaModal', () => {
 	} );
 
 	test( 'should show a Copy to media library button when viewing external media (no selection)', () => {
-		const tree = shallow(
+		const { container } = render(
 			<EditorMediaModal
 				{ ...baseProps }
 				view={ ModalViews.DETAIL }
 				setView={ spy }
 				selectedItems={ [] }
+				source="external"
 			/>
-		).instance();
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.getByTestId( 'media-modal-detail-base' ) ).toBeVisible();
+		expect( screen.getByRole( 'button', { name: /copy to media library/i } ) ).toBeVisible();
 	} );
 
 	test( 'should show a Copy to media library button when 1 external image is selected', () => {
-		const tree = shallow(
+		const { container } = render(
 			<EditorMediaModal
 				{ ...baseProps }
 				view={ ModalViews.DETAIL }
 				selectedItems={ DUMMY_MEDIA.slice( 0, 1 ) }
 				setView={ spy }
+				source="external"
 			/>
-		).instance();
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 2 );
+		expect( screen.getByRole( 'button', { name: /copy to media library/i } ) ).toBeVisible();
 	} );
 
 	test( 'should show a copy button when 1 external video is selected', () => {
-		const tree = shallow(
+		const { container } = render(
 			<EditorMediaModal
 				{ ...baseProps }
 				view={ ModalViews.DETAIL }
 				selectedItems={ DUMMY_VIDEO_MEDIA }
 				setView={ spy }
+				source="external"
 			/>
-		).instance();
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 2 );
+		expect( screen.getByRole( 'button', { name: /copy to media library/i } ) ).toBeVisible();
 	} );
 
 	test( 'should show a copy button when 2 or more external media are selected', () => {
-		const tree = shallow(
-			<EditorMediaModal { ...baseProps } view={ ModalViews.DETAIL } setView={ spy } />
-		).instance();
+		const { container } = render(
+			<EditorMediaModal
+				{ ...baseProps }
+				view={ ModalViews.DETAIL }
+				setView={ spy }
+				source="external"
+			/>
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 2 );
```

Comment:
Same as above with regards to `query*` vs `get*`.

```suggestion
		expect( screen.getAllByRole( 'button' ) ).toHaveLength( 2 );
```